### ISSUE TITLE
searching for series with colon in title not working in Komga extension

### ISSUE THREAD

**Issue by papaj-na-wrotkach** [tachiyomiorg/tachiyomi-extensions#16275]
### Source information

Komga

### Source language

ALL

### Steps to reproduce

Search for series with colon in title - eg. `Dance in the Vampire Bund II: Scarlet Order`

### Expected behavior

I should see the series (because I have it on my server)

### Actual behavior

No results shown

### Tachiyomi version

Preview r5745

### Android version

Android 11

### Other details

This is because
> Colon `:` is a special character to denote searching by field

as @gotson said in https://github.com/gotson/komga/issues/1101

Changing the code like this should fix it (enclose query with `"`)
```diff
 .../komga/src/eu/kanade/tachiyomi/extension/all/komga/Komga.kt  | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/all/komga/src/eu/kanade/tachiyomi/extension/all/komga/Komga.kt b/src/all/komga/src/eu/kanade/tachiyomi/extension/all/komga/Komga.kt
index 6bcaa7a7660..c312932f6bd 100644
--- a/src/all/komga/src/eu/kanade/tachiyomi/extension/all/komga/Komga.kt
+++ b/src/all/komga/src/eu/kanade/tachiyomi/extension/all/komga/Komga.kt
@@ -72,7 +72,7 @@ open class Komga(private val suffix: String = "") : ConfigurableSource, Unmetere
             else -> "series"
         }
 
-        val url = "$baseUrl/api/v1/$type?search=$query&page=${page - 1}&deleted=false".toHttpUrlOrNull()!!.newBuilder()
+        val url = "$baseUrl/api/v1/$type?search=\"$query\"&page=${page - 1}&deleted=false".toHttpUrlOrNull()!!.newBuilder()
 
         filters.forEach { filter ->
             when (filter) {
```

I applied the patch on my fork and it's working: [papaj-na-wrotkach/tachiyomi-extensions](https://github.com/tachiyomiorg/tachiyomi-extensions/commit/8a246546cb0e17d5c9f08612cb68babd026e3779)

This is very annoying, especially when migrating series after changing the address of the Komga server in the extension settings.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to version **[0.14.6](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/tachiyomiorg/tachiyomi/issues/new/choose).
- [X] I will fill out all of the requested information in this form.

**Comment by Seishirou101** [comment]
Considering that the extension mirrors komga in terms of searching, it would not be beneficial to change that as then it would prevent you when you want to search by field. You're probably better off just changing the entries themselves in komga without the `:` imo.

**Comment by papaj-na-wrotkach** [comment]
I really don't like to change titles. How about filtering special characters like `:` from title when trying to migrate a manga?

**Comment by Seishirou101** [comment]
No. As Gotson said in the very issue you linked. You will just have to just manually search, change the komga title, or do it yourself which you already have done.
